### DISCUSSION TITLE
Help to test card-form

### DISCUSSION THREAD

**Discussion by denn0n** [mercadopago/sdk-js#72] [General]
Hi after play some time whit the code provided on mercadopago an the one here I have  put this together but ? at the end i should get the json response for the processes payment but I'm not hahaha ! ups 

in the /payments I have 

`<form id="form-checkout" >
    <div id="form-checkout__cardNumber-container" class="container"></div> 
    <div id="form-checkout__expirationDate-container" class="container"></div> 
    <div id="form-checkout__securityCode-container" class="container"></div>
    <input type="text" name="cardholderName" id="form-checkout__cardholderName"/>
    <input type="email" name="cardholderEmail" id="form-checkout__cardholderEmail"/>
    <select name="issuer" id="form-checkout__issuer"></select>
    <select name="identificationType" id="form-checkout__identificationType"></select>
    <input type="text" name="identificationNumber" id="form-checkout__identificationNumber"/>
    <select name="installments" id="form-checkout__installments"></select>
    <button type="submit" id="form-checkout__submit">Pay</button>
    <progress value="0" class="progress-bar">loading...</progress>
    </form>
    <script src="https://sdk.mercadopago.com/js/v2"></script>`
`<script>
		
        const mp = new MercadoPago('TEST-a7f3b', {
            locale: 'en-US',
            advancedFraudPrevention: true,
        })

        const cardForm = mp.cardForm({
            amount: '100.5',
            iframe: true,
            autoMount: true,
            processingMode: 'aggregator',
            form: {
                id: 'form-checkout',
                cardholderName: {
                    id: 'form-checkout__cardholderName',
                    placeholder: 'Cardholder name',
                },
                cardholderEmail: {
                    id: 'form-checkout__cardholderEmail',
                    placeholder: 'Email',
                },
                cardNumber: {
                    id: 'form-checkout__cardNumber-container',
                    placeholder: 'Card number',
                },
                expirationDate: {
                    id: 'form-checkout__expirationDate-container',
                    placeholder: 'MM/YYYY'
                },
                securityCode: {
                    id: 'form-checkout__securityCode-container',
                    placeholder: 'CVV',
                },
                installments: {
                    id: 'form-checkout__installments',
                    placeholder: 'Total installments'
                },
                identificationType: {
                    id: 'form-checkout__identificationType',
                    placeholder: 'Document type'
                },
                identificationNumber: {
                    id: 'form-checkout__identificationNumber',
                    placeholder: 'Document number'
                },
                issuer: {
                    id: 'form-checkout__issuer',
                    placeholder: 'Issuer'
                }
            },
            callbacks: {
                onFormMounted: error => {
                    if (error) return console.warn('Form Mounted handling error: ', error)
                    console.log('Form mounted')
                },
                onFormUnmounted: error => {
                    if (error) return console.warn('Form Unmounted handling error: ', error)
                    console.log('Form unmounted')
                },
                onIdentificationTypesReceived: (error, identificationTypes) => {
                    if (error) return console.warn('identificationTypes handling error: ', error)
                    console.log('Identification types available: ', identificationTypes)
                },
                onPaymentMethodsReceived: (error, paymentMethods) => {
                    if (error) return console.warn('paymentMethods handling error: ', error)
                    console.log('Payment Methods available: ', paymentMethods)
                },
                onIssuersReceived: (error, issuers) => {
                    if (error) return console.warn('issuers handling error: ', error)
                    console.log('Issuers available: ', issuers)
                },
                onInstallmentsReceived: (error, installments) => {
                    if (error) return console.warn('installments handling error: ', error)
                    console.log('Installments available: ', installments)
                },
                onCardTokenReceived: (error, token) => {
                    if (error) return console.warn('Token handling error: ', error)
                    console.log('Token available: ', token)
                },
                onSubmit: (event) => {
                    event.preventDefault();
                    
                    const {
					paymentMethodId: payment_method_id,
					issuerId: issuer_id,
					cardholderEmail: email,
					amount,
					token,
					installments,
					identificationNumber,
                    paymentMethodId,
                    } = cardForm.getCardFormData();
                    
                    console.log('CardForm data available: ', cardForm.getCardFormData());
                
                    fetch ('/process_payment', {
                    method: 'POST',
                    headers: {
                    'Content-Type': 'application/json'
                    },
                    body: JSON.stringify({
                    token,
                    issuer_id,
                    payment_method_id,
                    transaction_amount: Number(amount),
                    installments: Number(installments),
                    description: 'product description',
                    payer: {
                    email,
                    identification: {
                    number: identificationNumber
					}
						}
							})
					})
                },
                onFetching:(resource) => {
                    console.log('Fetching resource: ', resource)

                    // Animate progress bar
                    const progressBar = document.querySelector('.progress-bar')
                    progressBar.removeAttribute('value')

                    return () => {
                        progressBar.setAttribute('value', '0')
                    }
                },
                onError: (error, event) => {
                    console.log(event, error);
                },
                onValidityChange: (error, field) => {
                    if (error) return error.forEach(e => console.log('${field}: ${e.message}'));
                    console.log('${field} is valid');
                },
                onReady: () => {
                    console.log("CardForm ready");
                }
            }
        })
    </script>`

and in /process_payment

`<?php
   require_once 'Mercado_Pago/vendor/autoload.php';
 
   MercadoPago\SDK::setAccessToken("TEST-5603");
 
   $payment = new MercadoPago\Payment();
   $payment->transaction_amount = (float)$_POST['transactionAmount'];
   $payment->token = $_POST['token'];
   $payment->description = $_POST['description'];
   $payment->installments = (int)$_POST['installments'];
   $payment->payment_method_id = $_POST['paymentMethodId'];
   $payment->issuer_id = (int)$_POST['issuer'];
 
   $payer = new MercadoPago\Payer();
   $payer->email = $_POST['email'];
   $payer->identification = array(
       "number" => $_POST['identificationNumber']
   );
   $payment->payer = $payer;
 
   $payment->save();
 
   $response = array(
       'status' => $payment->status,
       'status_detail' => $payment->status_detail,
       'id' => $payment->id
   );
   echo json_encode($response);
 
?>`

everything looks good in the console i get 

`(index):78 Form mounted
(index):142 Fetching resource:  identificationTypes
(index):86 Identification types available:  []
(index):160 CardForm ready
(index):157 ${field} is valid
(index):142 Fetching resource:  paymentMethods
(index):90 Payment Methods available:  [{…}]
(index):142 Fetching resource:  installments
(index):98 Installments available:  {merchant_account_id: null, payer_costs: Array(7)}
(index):157 ${field} is valid
(index):157 ${field} is valid
(index):157 ${field} is valid
(index):157 ${field} is valid
(index):157 ${field} is valid
(index):142 Fetching resource:  cardToken
(index):102 Token available:  {token: '00945690c9a00a421f'}
(index):118 CardForm data available:  {amount: '100.5', paymentMethodId: 'visa', token: '00945690c9a00a421f', issuerId: '159', installments: '1', …}amount: "100.5"cardholderEmail: "dtz@gmail.com"identificationNumber: "180032976"identificationType: ""installments: "1"issuerId: "159"merchantAccountId: ""paymentMethodId: "visa"processingMode: "aggregator"token: "00945690c9a00a421f"[[Prototype]]: Object` 

and there is not to much info about 
thanks !

**Comment by denn0n** [comment]
Sorry I did notice my mistake  ! how i can delete this discussion ?

**Reply by lucmantovani** [reply]
Hello @denn0n!

I am glad you managed to integrate with our products!
There is no need to delete this discussion!

Is there anything we could do to improve the documentation in order to let the implementation steps clear?
